lib/pbench/server/auth/__init__.py

```diff
@@ -50,7 +50,7 @@ def _method(
         self,
         method: str,
         path: str,
-        data: Union[Dict, None],
+        data: Union[Dict, str, None],
```

Comment:
This change feels wrong (see earlier comment(s)).

Ditto line 131.

Reply:
I think the type of 'data' is really automagically related to the content type, and from the caller point of view is fairly freeform, maybe even technically `Any`. If we're fully describing it, yeah, it could be a text string, a byte string, a dict (which on `data` rather than `json` is, as Webb said, considered to be form data), or `None` (no body).

I'm really tempted to say we should just type this as `typing.Any` and probably can't do much better. (And if we're going to explicitly type including `None` then it should be `Optional[]` and if we're including a dict than we should use `dict` rather than `typing.Dict`. So maybe `Optional[Union[dict, str]]` but `Any` would be my vote.

Reply:
See [my comment](https://github.com/distributed-system-analysis/pbench/pull/3235#discussion_r1107721971) below.